config/config_test.go

```diff
@@ -92,3 +100,109 @@ func TestNewDefault(t *testing.T) {
+
+	t.Setenv("XDG_CONFIG_HOME", home)
+
+	if err := config.CreatePaths(); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := os.Stat(config.Path()); err != nil {
+		if os.IsNotExist(err) {
+			t.Fatalf("config path '%v' not created", config.Path())
+		}
+		t.Fatal(err)
+	}
+
+	if _, err := os.Stat(filepath.Join(config.Path(), "repositories")); err != nil {
+		if os.IsNotExist(err) {
+			t.Fatalf("config path '%v' not created", config.Path())
+		}
+		t.Fatal(err)
+	}
+
+	// Trying to create when repositories path is invalid should error
+	_ = os.WriteFile("./invalidRepositoriesPath.txt", []byte{}, os.ModePerm)
+	t.Setenv("FUNC_REPOSITORIES_PATH", "./invalidRepositoriesPath.txt")
+	if err := config.CreatePaths(); err == nil {
+		t.Fatal("did not receive error when creating paths with an invalid FUNC_REPOSITORIES_PATH")
+	}
+
+	// Trying to Create config path should bubble errors, for example when HOME is
+	// set to a nonexistent path.
+	_ = os.WriteFile("./invalidConfigHome.txt", []byte{}, os.ModePerm)
+	t.Setenv("XDG_CONFIG_HOME", "./invalidConfigHome.txt")
+	if err := config.CreatePaths(); err == nil {
+		t.Fatal("did not receive error when creating paths in an invalid home")
+	}
+
+}
+
+// TestNewDefault_ConfigNotRequired ensures that when creating a new
+// config which would load a global config, its nonexistence causes no error.
+func TestNewDefault_ConfigNotRequired(t *testing.T) {
+	// Custom config home results in a config file default path of
+	// ./testdata/func/config.yaml
+	home, cleanup := Mktemp(t)
+	t.Cleanup(cleanup)
+	t.Setenv("XDG_CONFIG_HOME", home)
+
+	_, err := config.NewDefault() // Should not error despite no config.
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestRepositoriesPath returns the path expected
+// (XDG_CONFIG_HOME/func/repositories by default)
+func TestRepositoriesPath(t *testing.T) {
+	home, cleanup := Mktemp(t)
+	t.Cleanup(cleanup)
+	t.Setenv("XDG_CONFIG_HOME", home)
+
+	expected := filepath.Join(home, "func", config.Repositories)
+	if config.RepositoriesPath() != expected {
+		t.Fatalf("unexpected reposiories path: %v", config.RepositoriesPath())
+	}
+}
+
+// TestWrite ensures that a config is written to the given path and errors
+// are returned correctly when the path does not exist.
+func TestWrite(t *testing.T) {
+	home, cleanup := Mktemp(t)
+	t.Cleanup(cleanup)
+	t.Setenv("XDG_CONFIG_HOME", home)
+
+	cfg := config.New()
+	cfg.Language = "example"
+
+	// First try writing to a nonexistent path
+	if err := cfg.Write(config.ConfigPath()); err == nil {
+		t.Fatal("did not receive error writing to a nonexistent path")
+	}
+
+	// Create the path and try again
+	if err := config.CreatePaths(); err != nil {
+		t.Fatal(err)
+	}
+	if err := cfg.Write(config.ConfigPath()); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestDefaultNamespace ensures that, when thre is a problem determining the
+// active namespace, the static DefaultNamespace ("default") is used.
+func TestDefaultNamespace(t *testing.T) {
```

Comment:
Might be nice to also have a test with mocked `KUBECONFIG` containing `ns` set.

Reply:
Yes, apparently I agreed with you already.  This test is much [more fully fleshed out](https://github.com/knative/func/pull/1353/files#diff-373f5578b8f362a364b50b0d01919a57da64c128547ca10d49df8ae422237c82R201) in #1353.  Let me know if that's not what you were thinking.